File: 240227/1이 3개 이상 있는 위치/place-more-than-3-ones.cpp
#include <iostream>
using namespace std;


int arr[101][101];
int main() {
    int dx[4] = {1,0,-1,0};
    int dy[4] = {0,-1,0,1};
    int n;
    cin >> n;
    for (int i = 0; i < n; i++)
    {
        for (int j = 0; j < n; j++)
        {
           cin >> arr[i][j];
        }
    }
    int x, y;
    int tx, ty;
    int three = 0;
    int cnt = 0;
    for (int i = 0; i < n; i++)
    {
        for (int j = 0; j < n; j++)
        {
           x = i;
           y = j;
           for (int k = 0; k < 4; k++)
           {
                tx = x + dx[k];
                ty = y + dy[k];
                if ((0<=tx && tx <= n && 0 <= ty && ty <= n) && arr[tx][ty]==1)
                {
                    three++;
                }
           }
           if (three >= 3) cnt++;
           three = 0;
        }
    }

    cout << cnt;

    return 0;
}

File: 240316/1이 3개 이상 있는 위치/place-more-than-3-ones.py
n = int(input())

arr = []

for i in range(n):
    arr.append(list(map(int, input().split())))

# 상하좌우
dx = [-1, 1, 0, 0]
dy = [0,0,-1,1]

def in_range(x):
    if x>=0 and x<n:
        return True
    return False

def three_ones(x, y):
    cnt=0
    for xx, yy in zip(dx, dy):
        r = x+xx
        c = y+yy
        if in_range(r) and in_range(c):
            if arr[r][c]==1:
                cnt+=1
                if cnt>=3:
                    return True

    return False

count=0
for i in range(n):
    for j in range(n):
        if three_ones(i,j):
            count+=1
print(count)